### PR TITLE
Kodi 17

### PR DESCRIPTION
**Pull request by lrusak**
Just putting this in place for after we fork LibreELEC-7.0

### CODE REVIEW

File: packages/mediacenter/kodi/package.mk [hunk 1]
```diff
@@ -404,6 +403,22 @@ post_makeinstall_target() {
       cp $PKG_DIR/config/appliance.xml $INSTALL/usr/share/kodi/system/settings
     fi
 
+  # update addon manifest
+  xmlstarlet ed -L -d "/addons/addon[text()='service.xbmc.versioncheck']" \
+    $INSTALL/usr/share/kodi/system/addon-manifest.xml || :
+  xmlstarlet ed -L -d "/addons/addon[text()='skin.estouchy']" \
+    $INSTALL/usr/share/kodi/system/addon-manifest.xml || :
+  xmlstarlet ed -L --subnode "/addons" -t elem -n "addon" -v "os.openelec.tv" \
```
Comment (MilhouseVH) [inline packages/mediacenter/kodi/package.mk]: Shouldn't this and the next two be rebranded?

File: distributions/OpenELEC/options [hunk 2]
```diff
@@ -22,11 +22,11 @@
 
 # Skins to install (Confluence)
```
Comment (MilhouseVH) [inline distributions/OpenELEC/options]: Heh, this should be distributions/LibreELEC/options as well.

File: packages/mediacenter/kodi-theme-Estuary/patches/kodi-theme-Estuary-0001-add_oe_settings_to_homescreen.patch [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
++++ b/1080i/Home.xml	2016-03-02 23:17:43.347966012 +0100
+@@ -1089,6 +1089,12 @@
+ 							<property name="id">weather</property>
+ 							<visible>!Skin.HasSetting(HomeMenuNoWeatherButton)</visible>
+ 						</item>
++						<item>
++							<label>@DISTRONAME@</label>
++							<onclick>RunAddon(service.openelec.settings)</onclick>
++							<thumb>icons/home/openelec.png</thumb>
++							<property name="id">openelec</property>
```
Comment (MilhouseVH) [inline packages/mediacenter/kodi-theme-Estuary/patches/kodi-theme-Estuary-0001-add_oe_settings_to_homescreen.patch]: Need to re-brand this too :)

File: packages/mediacenter/kodi-binary-addons/inputstream.mpd/package.mk [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+configure_target() {
+  cmake -DCMAKE_TOOLCHAIN_FILE=$CMAKE_CONF \
+        -DCMAKE_INSTALL_PREFIX=/usr \
+        -DCMAKE_MODULE_PATH=$SYSROOT_PREFIX/usr/lib/kodi \
+        -DCMAKE_PREFIX_PATH=$SYSROOT_PREFIX/usr \
+        ..
+}
+
+post_makeinstall_target() {
+  mkdir -p wv && cd wv
```
Comment (MilhouseVH) [inline packages/mediacenter/kodi-binary-addons/inputstream.mpd/package.mk]: This won't work unless we're including the wvdecrypter code too (required by libwidevine).

Comment (lrusak) [inline packages/mediacenter/kodi-binary-addons/inputstream.mpd/package.mk]: Where does that come from.?

**Comment by MilhouseVH** [comment]
Some extras we will need:

peripheral.joystick (needs adding to addon-manifest.xml too, along with inputstream.mpd too)

There's a number of ffmpeg and libbluray patches from popcornmix we need - I think I'll add these in a separate PR

File: packages/mediacenter/kodi-binary-addons/inputstream.mpd/package.mk (hunk 4 above)
Comment (MilhouseVH) [inline packages/mediacenter/kodi-binary-addons/inputstream.mpd/package.mk]: That's currently in the same package that includes libwidevinecdm.so - in my own builds I merge it together with inputstream.mpd in which case this package.mk would work.

File: packages/mediacenter/kodi/package.mk [hunk 5]
```diff
@@ -406,6 +405,22 @@ post_makeinstall_target() {
       cp $PKG_DIR/config/appliance.xml $INSTALL/usr/share/kodi/system/settings
     fi
 
+  # update addon manifest
+  xmlstarlet ed -L -d "/addons/addon[text()='service.xbmc.versioncheck']" \
+    $INSTALL/usr/share/kodi/system/addon-manifest.xml || :
```
Comment (MilhouseVH) [inline packages/mediacenter/kodi/package.mk]: Can you remove `|| :` from the `xmlstarlet` calls as it shouldn't fail silently - thanks.

**Comment by lrusak** [comment]
Updated. I'll still have to sort out inputstream.mpd

File: packages/mediacenter/kodi-binary-addons/inputstream.mpd/package.mk [hunk 6]
```diff
@@ -46,9 +46,6 @@ post_makeinstall_target() {
 
   mkdir -p $INSTALL/usr/lib/kodi/addons/$PKG_NAME/decrypter
   cp -P $ROOT/$PKG_BUILD/.$TARGET_NAME/wv/libssd_wv.so $INSTALL/usr/lib/kodi/addons/$PKG_NAME/decrypter
-  if [ -d $ROOT/$PKG_BUILD/$TARGET_ARCH ]; then
-    cp -P $ROOT/$PKG_BUILD/$TARGET_ARCH/* $INSTALL/usr/lib/kodi/addons/$PKG_NAME/decrypter
-  fi
```
Comment (MilhouseVH) [inline packages/mediacenter/kodi-binary-addons/inputstream.mpd/package.mk]: We'll need a sym link here, too:
```
ln -sf /storage/decrypters/libwidevinecdm.so $INSTALL/usr/lib/kodi/addons/$PKG_NAME/decrypter/libwidevinecdm.so
```

File: packages/mediacenter/p8-platform/package.mk [hunk 7]
```diff
@@ -16,18 +16,19 @@
...
-PKG_VERSION="081032f"
+PKG_NAME="p8-platform"
+PKG_VERSION="38343e0"
 PKG_REV="1"
 PKG_ARCH="any"
 PKG_LICENSE="GPL"
 PKG_SITE="http://www.kodi.tv"
-PKG_URL="$DISTRO_SRC/$PKG_NAME-$PKG_VERSION.tar.xz"
+PKG_URL="https://github.com/Pulse-Eight/platform/archive/$PKG_VERSION.tar.gz"
+PKG_SOURCE_DIR="platform-$PKG_VERSION*"
```
Comment (MilhouseVH) [inline packages/mediacenter/p8-platform/package.mk]: `PKG_SOURCE_DIR="platform-$PKG_VERSION*"` is not required (and `platform-$PKG_VERSION*` is also incorrect).

Comment (lrusak) [inline packages/mediacenter/p8-platform/package.mk]: try it without it :)

**Comment by MilhouseVH** [comment]
We will need to apply the following patch

http://nmacleod.com/public/oebuild/patches/00_fix_le_estuary.txt

when using the latest Kodi master (after https://github.com/xbmc/xbmc/pull/9488) - just a change to the indentation level.

**Pull request closed without merging**